### ISSUE TITLE
Youtube: External Links are stores as oembed

### ISSUE THREAD

**Issue by PascalMinder** [wallabag/wallabag#3638]
When I add a youtube link (for [example](url=https://www.youtube.com/watch?v=SS7z8WsXPMk)) the external link stores is `https://www.youtube.com/oembed?format=xml&url=https://www.youtube.com/watch?v=SS7z8WsXPMk`.

Is that behavior intended?

### Environment

* wallabag version: 2.3.2

**Comment by sanpii** [comment]
This "feature" annoys me because I can’t directly copy/paste the link to youtube-dl.

**Comment by sigmap** [comment]
Can confirm this happens on 2.3.3

**Comment by GeorgeHahn** [comment]
Is there any way to get the original youtube link through the API? I am running into an issue with this behavior in a [wallabag reducer](https://github.com/GeorgeHahn/wallabag_reducer) plugin to automatically download youtube links.

**Comment by Kdecherf** [comment]
@sanpii @GeorgeHahn as a workaround, starting with 2.3.4 the original url is stored in the field `origin_url`, see https://github.com/wallabag/wallabag/pull/3553

**Comment by N07070** [comment]
Maybe the link to the youtube video should be the origin each time ? I don't understand the point of having the oembed tag code in the URL.

Otherwise, this issue could be closed ?

**Comment by bhigginsuk** [comment]
This started causing me problems last week, any YouTube video I save gets stored as an oembed and consequently doesn't work and doesn't link to the actual video. Here is a screenshot showing a video I saved on 24th Feb which was working fine, and one on the 28th Feb which is not (none have worked since).

![Broken YouTube videos](https://user-images.githubusercontent.com/14295333/109715854-0f7e9280-7b9c-11eb-8c9b-f9ed2a3d9f33.png)

I'm not sure whether this is a separate bug, but it makes YouTube videos unplayable on both desktop and mobile. Opening the link externally ([example](https://www.youtube.com/oembed?format=xml&url=https://www.youtube.com/watch?v%3DWah1DbFVFiY)) just loads an XML document.
![oembed XML](https://user-images.githubusercontent.com/14295333/109716163-62584a00-7b9c-11eb-9451-62cc07af2b71.png)

The only way I can play YouTube videos now is to copy the title and paste it in the YouTube search.

**Comment by mx-psi** [comment]
Maybe Wallabag should treat Youtube links in a special way and change the `oembed` URL by the actual URL automatically? I don't think the average user benefits from having the `oembed` URL in any way.

**Comment by gotmax23** [comment]
It might also be nice to have the saved YouTube video embeded directly in Wallabag.

**Comment by bhigginsuk** [comment]
This appears to now be fixed, my YouTube videos are working as expected.

**Comment by scharmach** [comment]
> 
> 
> This appears to now be fixed, my YouTube videos are working as expected.

I'm running the latest version 2.4.2 and YouTube-videos are still not working as expected. I don't see any commit since then that would suggest otherweise either: https://github.com/wallabag/wallabag/compare/2.4.2...master

Thumbnails still don't show and links are still in this oembed-format. So what do you mean?

**Comment by bhigginsuk** [comment]
I take it back, still an issue. I mostly use Wallabag for sending articles from my desktop to my phone, so don't open the web app very often. Just the other day my Android phone started handling videos normally again; i.e. allowing me to play them in-app and link to an external app to handle it.

I must confess I didn't actually check any further than this. I can confirm it's still not working in the browser for me. Sorry for the misleading comment.

**Comment by ntnsndr** [comment]
I've never been happy with the Wallabag handling of YT links due to the oembed issue above. But in recent weeks, YT hasn't been working at all. Now, all YT links appear in Wallabag as [this website](https://consent.youtube.com/ml?continue=https://www.youtube.com/watch?v=tGwZwLd_KYk&gl=DE&hl=de&pc=yt&uxe=23983172&src=1). Is this something specific to wallabag.it, or to the software itself? Is there something I can do to prevent this?

I am a paying wallabag.it user.

**Comment by dominiksta** [comment]
Is anybody working on this? This bug is especially annoying on android, since there doesn't seem to be a way to open the `original_url` field at all, which renders wallabag completely unusable for youtube links.

**Comment by Kdecherf** [comment]
@dominiksta as far as I know there's no progress on this one.

**Comment by eb0687** [comment]
Was this issue ever fixed? I still seem to get the same oembed error as others have mentioned here..

**Comment by bcutter** [comment]
https://github.com/wallabag/wallabag/issues/6567#issuecomment-2993692488 + https://github.com/wallabag/wallabag/issues/2149

**Comment by nicosomb** [comment]
I was playing with youtube videos in wallabag and one fix for this issue could be to skip the step where we set the origin URL in `src/Helper/ContentProxy.php`, like @N07070 suggested. 

@j0k3r @Kdecherf @yguedidi what do you think?

**Comment by yguedidi** [comment]
Yes I think it makes sense to keep the original Youtube link as the entry URL. Then we can have a special processing to set the content of the entry, maybe using a library like [embed/embed](https://packagist.org/packages/embed/embed)
